Test/pipeline converstion

alexvy86 opened the pull request:
## Description

Testing move of the `Build - client packages` pipeline to new base template.

## Reviewer Guidance

The review process is outlined on [this wiki page](https://github.com/microsoft/FluidFramework/wiki/PR-Guidelines#guidelines).

[AB#5816](https://dev.azure.com/fluidframework/235294da-091d-4c29-84fc-cdfc3d90890b/_workitems/edit/5816)

ghoshkaj committed:
test-build-client-pipeline-1es-yaml-conversion (#18238)

[How contribute to this
repo](https://github.com/microsoft/FluidFramework/blob/main/CONTRIBUTING.md).

[Guidelines for Pull
Requests](https://github.com/microsoft/FluidFramework/wiki/PR-Guidelines#guidelines).

The sections included below are suggestions for what you may want to
include.
Feel free to remove or alter parts of this template that do not offer
value for your specific change.

## Description

> A concise description of the changes (bug or feature) and their
impact/motivation.
> If this description is short enough to be used as the title, delete
this section and just use the title.

> For bug fixes, also include specifics of how to reproduce it / confirm
it is fixed.

> If this Pull Request should close/resolve any issues when merged, use
[the special syntax for
that](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword)
here.

## Breaking Changes

> If this introduces a breaking change, please describe the impact and
migration path for existing applications below.
> See
[Breaking-vs-Non-breaking-Changes](https://github.com/microsoft/FluidFramework/wiki/Breaking-vs-Non-breaking-Changes)
for details.
> If there are no breaking changes, delete this section.

## Reviewer Guidance

The review process is outlined on [this wiki
page](https://github.com/microsoft/FluidFramework/wiki/PR-Guidelines#guidelines).

> List any specific things you want to get reviewer opinions on, and
anything a reviewer would need to know to review this PR effectively.
> Things you might want to include:
>
> - Questions about how to properly make automated tests for your
changes.
> -   Questions about design choices you made.
> - Descriptions of how to manually test the changes (and how much of
that you have done).
> -   etc.
>
> If you have any questions in this section, consider making the PR a
draft until all questions have been resolved.

ghoshkaj committed:
prettify the yaml

ghoshkaj committed:
update buildclient

ghoshkaj committed:
move variables to its own file

ghoshkaj committed:
move where variables is pulled in from

ghoshkaj committed:
update variables file

ghoshkaj committed:
add arrow function

ghoshkaj committed:
indent

ghoshkaj committed:
move arrow to M635 template

ghoshkaj committed:
add 1es to PublishPipelineArtifact@1

ghoshkaj committed:
use oour own pools

ghoshkaj committed:
use linux shared pools

ghoshkaj committed:
keep 1es validation as windows and use linux for our builds

ghoshkaj committed:
use Azure-Pipelines-1ESPT-ExDShared pool

ghoshkaj committed:
check if bash script breaks

ghoshkaj committed:
put back build job pool to be linux

ghoshkaj committed:
try using sourceAnalysisPool for sdl

ghoshkaj committed:
force NewLarge and linux os

ghoshkaj committed:
move pool out of build

ghoshkaj committed:
use new pool

ghoshkaj committed:
add sbom path

ghoshkaj committed:
turn off sbom because its not required

ghoshkaj committed:
use NewLarge for build-client as well

ghoshkaj committed:
remove all cosmetic changes in build-client.yml

ghoshkaj committed:
revent lint changes

ghoshkaj committed:
revert include-git-tag-steps

ghoshkaj committed:
include-install-pnpm.yml

ghoshkaj committed:
include-install.yml

ghoshkaj committed:
include-policy-check

ghoshkaj committed:
include-process-test-results.yml

ghoshkaj committed:
include-publish-npm-package-deployment

ghoshkaj committed:
include-publish-npm-package-steps.yml

ghoshkaj committed:
include-set-package-version.yml

ghoshkaj committed:
include-telemetry-setup.yml

ghoshkaj committed:
include-telemetry-setup.yml

ghoshkaj committed:
include-test-task.yml

ghoshkaj committed:
include-use-node-version.yml

ghoshkaj committed:
include-vars.yml

ghoshkaj committed:
remove isReleaseGroup

ghoshkaj committed:
include missing checks

ghoshkaj committed:
see if build-client is the reason for failure

ghoshkaj committed:
revert isReleaseGroup: true

rohandubal committed:
Port PR changes from (#18250) specific to templates directory

rohandubal committed:
Update all templates to use new Small-1ES pools instead of previous Small

rohandubal committed:
Create separate folder for 1ES templates

rohandubal committed:
Update build-client pipeline and 1ES templates to strictly use 1ES templates

rohandubal committed:
Replicate files from 1ES folder to root templates folder to simplify diff/review

rohandubal committed:
Refine build-npm-package

rohandubal committed:
More udpates to build-npm-package.yml

rohandubal committed:
Fix string typo in build-npm-package.yml

rohandubal committed:
Fix string typo in build-npm-package.yml

rohandubal committed:
Fix string typo in build-npm-package.yml

rohandubal committed:
Use dynamic arrow connection name

rohandubal committed:
Use if-else to determine arrow service connection name

rohandubal committed:
Fix formatting issues coming from 1ES tool

alexvy86 committed:
Centralize variable inclusions in build-npm-package (#18385)

alexvy86 committed:
Merge branch 'main' into test/pipeline-converstion

alexvy86 committed:
Resolve variables at runtime

rohandubal committed:
Convert build-api-markdown-documenter.yml pipeline

rohandubal committed:
Replicate templates in 1ES folder

rohandubal committed:
Fix readonly parameter in build-npm-package

rohandubal committed:
Convert build-benchmark-tool.yml for 1ES

rohandubal committed:
Revert AI made changes by 1ES tool in include-publish-npm-package-steps.yml

rohandubal committed:
Convert build-protocol-definitions.yml pipeline

rohandubal committed:
Revert un-needed changes from the tool

rohandubal committed:
Migrate server-historian.yml and replicate 1ES folder templates to sync

rohandubal committed:
Fix typo in includ-vars.yml

rohandubal committed:
Fix indentation issue related to variables section in include-publish-npm-package.yml

rohandubal committed:
Remove variables from include-publish-npm-package.yml

rohandubal committed:
Attempt to fix parameters in include-publish-npm-package.yml

rohandubal committed:
Migrate server-historian.yml

rohandubal committed:
Migrate build-azure.yml

rohandubal committed:
Fix tool issues in build-azure.yml

rohandubal committed:
Use 1ES poolbuild for build-azure

rohandubal committed:
Merge branch 'main' into test/pipeline-converstion

rohandubal committed:
Manually pick changes from main to apply in pipeline templates; esp build-npm-package and include-publish-npm-package.yml

rohandubal committed:
fix typo in build-npm-package

rohandubal committed:
fix typo in build-npm-package

rohandubal committed:
Fix merge issues for build-npm-package

alexvy86 commented on tools/pipelines/templates/include-generate-notice-steps.yml:
I'd say revert this file... all the tool did in this file is remove the trailing newline, which we should keep.

Same for `include-git-tag-steps.yml` and `include-install-pnpm.yml` (there it also removed comments, so more reason for reverting).

alexvy86 commented on tools/pipelines/templates/include-policy-check.yml:
```diff
@@ -28,14 +28,14 @@ stages:
     displayName: Run checks
     steps:
-    - template: include-use-node-version.yml
+    - template: /tools/pipelines/templates/1ES/include-use-node-version.yml@self
```
Not sure the non-1ES templates should include the 1ES ones or run in the 1ES pools... unless this is already preparation to move everything to the new templates, but in that case maybe let's just update all the templates inline at their current location?

alexvy86 commented on tools/pipelines/templates/include-install.yml:
```diff
@@ -25,4 +25,4 @@ steps:
       script: |
-        ${{ parameters.packageManagerInstallCommand }}
\ No newline at end of file
+        ${{ parameters.packageManagerInstallCommand }}
```
Restore trailing newline

alexvy86 reviewed (commented):
Left a few comments; I have more similar ones but this is still draft so might be early to comment on everything.

We've already talked about restoring comments and newlines removed by the tool... I'd also suggest we restore the trailing newlines it removed from the last line in some files.

rohandubal committed:
Update template paths to use 1ES folder for converted/migrated pipelines

rohandubal committed:
Update tools/pipelines/build-api-markdown-documenter.yml

Co-authored-by: Alex Villarreal <716334+alexvy86@users.noreply.github.com>

seanimam committed:
Convert server-gitssh.yml pipeline

seanimam committed:
adds back newline at the end of server-gitssh.yml

seanimam committed:
Converts server-gitrest.yml pipeline

seanimam committed:
Convert build-test-tools.yml pipeline

seanimam committed:
Converts build-build-common.yml pipeline

seanimam committed:
converts build-build-tools.yml pipeline

seanimam committed:
converts build-eslint-config-fluid.yml pipeline

seanimam committed:
converts repo-policy-check.yml pipeline

seanimam committed:
Converts build-markdown-magic.yml pipeline

seanimam committed:
converts build-common-utils.yml pipeline

seanimam committed:
Converts build-common-definitions.yml pipeline

seanimam committed:
attempt to convert build-docker-service.yml template

seanimam committed:
updates the build-docker-service file to the converted version within the 1ES folder

seanimam committed:
temporary line addition to test an error in include-publish-docker-service-steps.yml

seanimam committed:
updates Docker@0 to Docker@1 in include-publish-docker-service-steps

seanimam committed:
updates include-publish-docker-service.yml to use 1ES template instead of the original

seanimam committed:
Updates templates include-publish-docker-service-step  in both 1ES and original folder which was missed the first time

seanimam committed:
updates include-publish-docker-service-steps.yml to reference the 1ES folder the for the include-git-tags template

seanimam committed:
one more missing reference to 1ES folder

rohandubal committed:
Merge branch 'main' into test/pipeline-converstion

rohandubal committed:
Merge branch 'main' into test/pipeline-converstion

rohandubal commented on tools/pipelines/templates/include-install-pnpm.yml:
```diff
@@ -39,13 +39,10 @@ steps:
   inputs:
     targetType: 'inline'
     workingDirectory: ${{ parameters.buildDirectory }}
-    # The previous task (cache restoration) can timeout, which is classified as canceled, but since it's just cache
```
Need to bring this comment in.

rohandubal commented on tools/pipelines/templates/include-install-pnpm.yml:
```diff
@@ -39,13 +39,10 @@ steps:
     targetType: 'inline'
     workingDirectory: ${{ parameters.buildDirectory }}
-    # The previous task (cache restoration) can timeout, which is classified as canceled, but since it's just cache
-    # restoration, we want to continue even if it timed out.
     condition: or(succeeded(), canceled())
-    # workspace-concurrency 0 means use use the CPU core count. This is better than the default (4) for larger agents.
```
Same here, need to bring this comment in.

rohandubal commented on tools/pipelines/templates/include-use-node-version.yml:
```diff
@@ -1,14 +1,9 @@
-# Copyright (c) Microsoft Corporation and contributors. All rights reserved.
-# Licensed under the MIT License.
-
-# include-use-node-version template to use a specific node version
-
```
Missed reverts, need to get back comments and license header

rohandubal commented on tools/pipelines/templates/include-use-node-version.yml:
```diff
@@ -1,14 +1,9 @@
+# This Yaml Document has been converted by ESAI Yaml Pipeline Conversion Tool.
+# Please make sure to check all the converted content, it is your team's responsibility to make sure that the pipeline is still valid and functions as expected.
 parameters:
-# The node version required. The version must already be installed in the image.
```
Another one

rohandubal committed:
Update build-npm-package template to use inline publish artifact steps

rohandubal committed:
Fix missing inputs tag for publishpipelineartifact step

alexvy86 commented on tools/pipelines/templates/1ES/build-npm-package.yml:
```diff
@@ -426,11 +429,12 @@ extends:
-                      targetPath: ${{ parameters.buildDirectory }}/artifacts/bundleAnalysis
-                      artifactname: bundleAnalysis
-                      publishLocation: Container
+                  inputs:
```
Indentation here is funky. Also, displayName and condition should be at the same level as inputs, IIRC; the rest are children of inputs.

rohandubal committed:
Fix indentation for inputs tag for publishpipelineartifact step

seanimam committed:
Updates Docker@0 to Docker@1 with new inputs in build-docker-service.yml

seanimam committed:
refactors tagging arguements under 1ES.PushContainerImage@1 and 1ES.BuildContainerImage@1 to do tagging using buildArguments instead

seanimam committed:
Updates a few Docker@1 commands and 1ES.Push commands method of tagging

seanimam committed:
keeping both copies of build-docker-service in sync

seanimam committed:
Removes unecessary tag argument

alexvy86 committed:
Fix indentation and remove comments

seanimam committed:
removes Docker cleanup commands from build-docker-service.yml

seanimam committed:
applies previous commit to remove docker cleanup commands to 1ES copy of build-docker-service.yml

seanimam committed:
Adds container registry url into the 'image' property of various docker tasks in build-docker-service.yml

alexvy86 commented on tools/pipelines/templates/1ES/build-docker-service.yml:
```diff
@@ -415,7 +415,7 @@ extends:
             displayName: Docker Push
             inputs:
-              image: ${{ variables.buildContainerName }}:$(containerTagSuffix)
+              image: $(containerRegistryUrl)/${{ variables.buildContainerName }}:$(containerTagSuffix)
```
I think `buildContainerName` already has a "prefix" that acts as registry. See how we define it [here](https://github.com/microsoft/FluidFramework/blob/a3dfbb6ac3c7c8893734907eaeef13d03b149d2b/tools/pipelines/templates/build-docker-service.yml#L108-L113).

seanimam replied:
Updated.

seanimam committed:
updates image parameter value

seanimam committed:
Updates syntax for referencing variables to use runtime computation systax instead of compile time

alexvy86 commented on tools/pipelines/templates/1ES/build-docker-service.yml:
```diff
@@ -403,20 +410,20 @@ extends:
         # Build
         - task: 1ES.BuildContainerImage@1
-          displayName: Docker Build - ${{ variables.buildContainerName }}
+          displayName: Docker Build - $(buildContainerName)
```
Let's see what happens here, not sure how this will act on a display name.

alexvy86 reviewed (commented):
The changes you made look good... but I think there's still some lines that we'll need to change. E.g. line 143 (and a few other instances right after this) is still trying to use variables at compile time:

```
        - ${{ if eq(variables.testBuild, true) }}:
          - name: buildContainerName
            value: test/${{ parameters.containerName }}
        - ${{ if eq(variables.testBuild, false) }}:
          - name: buildContainerName
            value: build/${{ parameters.containerName }}
```

Or line 272:

`- ${{ if eq(variables.componentDetection, true) }}:`

Line 276 does have a fix, I believe. Instead of 

`requireNotice: ${{ eq(variables.releaseImage, true) }}`

Use

`requireNotice: $[eq(variables.releaseImage, true)]`

alexvy86 committed:
Move variables template to root of server pipelines

alexvy86 committed:
Fix quote location

alexvy86 committed:
Print more variables

alexvy86 committed:
Testing with parameters

alexvy86 committed:
Cleanup debugging prints

alexvy86 committed:
Use compile-time pushImage. Add debugging prints. Make variable dependencies more tractable.

alexvy86 committed:
Fix condition syntax

alexvy86 committed:
Condition syntax fix, attempt 2

alexvy86 committed:
Syntax? Try to get build number

alexvy86 committed:
Use buildkit

alexvy86 committed:
Enable network when building in docker container

alexvy86 committed:
Add new parameters to other server pipelines

alexvy86 committed:
Use correct values

alexvy86 committed:
Authentication to ACR

alexvy86 committed:
Try different syntax

alexvy86 committed:
Hardcode for tests

seanimam committed:
converts server-routerlicious

seanimam committed:
removes old accidental additonal of variables section to server-routerlicious

alexvy86 committed:
Restore changes in server-routerlicious

alexvy86 committed:
Try to conditionally set containerRegistryConnection

alexvy86 committed:
Syntax?

alexvy86 committed:
New attempt at conditional containerRegistryConnection

alexvy86 committed:
Attempt 3

alexvy86 committed:
Evaluate containerRegistryConnection at compile time

alexvy86 committed:
Try a mix of compile-time and runtime variables

alexvy86 committed:
Fix parameter I missed

rohandubal committed:
Revert changes to use output section for pipeline artifacts

alexvy86 committed:
Try moving container-registry-info to the shared template

alexvy86 committed:
Revert "Try moving container-registry-info to the shared template"

This reverts commit 0441e09449dfee11f92d4b1fb539c95b69fec642.

alexvy86 committed:
Add container-registry-info to other server pipelines

alexvy86 committed:
Add variables to server-gitssh

alexvy86 committed:
Remove extra parameters in server-routerlicious

alexvy86 committed:
Put all artifacts under the same folder

alexvy86 committed:
Apply changes to other copy of template

alexvy86 committed:
Fix more compile-time/runtime issues in server pipelines

alexvy86 committed:
Small cleanup

alexvy86 committed:
Attempt at converting include-publish-docker-service-steps

alexvy86 committed:
Remove stage duplicated by mistake

alexvy86 committed:
Attempt to bring back ComponentGovernance job

alexvy86 committed:
Attempt 2 - will variables defined in the pipeline be available statically in extended templates?

alexvy86 committed:
Fix use of variables for compile-time conditional expressions.

componentDetection is just given the same value as publish, so no need to pass a separate parameter.

alexvy86 committed:
Fully remove unnecessary componentDetection variable

Its value is just that of publish, which we already have to do a dance for passing around. No need to do it for two things whose value is always the same.

alexvy86 committed:
Fix parameter name in debugging echo

alexvy86 committed:
More compile-time variable fixes.

The 'buildBase' variable is kind of redundant; it's always true for all server pipelines except gitssh (which also doesn't provide any of the related test/lint/pack/docs parameters), so it was easier to get rid of it entirely and special case gitssh. 'testCoverage' was only used for one purpose so inlining it was the easiest solution.

alexvy86 committed:
Remove unnecessary printed variable

alexvy86 committed:
Add more troubleshooting prints

alexvy86 committed:
Fix base container build. Sort (and add) troubleshooting prints.

alexvy86 committed:
Fix usage of Docker@1 run

alexvy86 committed:
Keep `docker run` tasks in foreground so ADO has the output

alexvy86 committed:
Enable use of cache for docker builds

alexvy86 committed:
Restore pnpm store pruning for server pipelines

alexvy86 committed:
Alphabetize

alexvy86 committed:
Increase test timeout

alexvy86 committed:
Merge branch 'main' into test/pipeline-converstion

alexvy86 committed:
Fix server-routerlicious so it runs in the Large-1ES pool

alexvy86 committed:
Fix for permissions problem in 1ES credscan

alexvy86 committed:
Fix pool name for server-routerlicious

alexvy86 committed:
Try running tests without --user root

rohandubal committed:
Add cred scan suppresions which cause false positive from 1espt scanner

alexvy86 committed:
Try to update file permisisons to make CredScan happy

alexvy86 committed:
sudo you will obey

alexvy86 committed:
Prettier on CredScanSuppressions.json

alexvy86 committed:
Copy flie by hand

alexvy86 committed:
Fix copy of tinylicious log file

alexvy86 committed:
Create folder for tinylicious log

alexvy86 committed:
Parameter back into variable. Not used at compile time.

alexvy86 committed:
Update some pipelines to latest requirements of build-npm-package

alexvy86 committed:
Fix conditions and conditionals for copying t9s log

alexvy86 committed:
Move repo-policy-check back to Small pool

alexvy86 committed:
Update other build pipelines for latest build-npm-packages requirement

alexvy86 committed:
Fix syntax in conditional

alexvy86 committed:
Merge branch 'main' into test/pipeline-converstion

alexvy86 committed:
Fix printing of debugging values

alexvy86 committed:
Fixes to pools used

alexvy86 committed:
Update pools for two pipelines

alexvy86 committed:
Comment

alexvy86 committed:
Add new required parameter

alexvy86 committed:
Update template paths

alexvy86 committed:
Hoist the FORCE_COLOR variable to include-vars

alexvy86 committed:
Update pipeline to use 1es template

alexvy86 committed:
Merge branch 'main' into test/pipeline-converstion

alexvy86 committed:
Update templates in 1ES folders to match the ones outside

alexvy86 committed:
Merge branch 'main' into test/pipeline-converstion

alexvy86 commented:
This work has been completed in other PRs, closing initial draft.
